torch/csrc/profiler/util.cpp

```diff
@@ -365,12 +366,18 @@ std::unordered_map<std::string, std::string> saveNcclMeta(
       kDtype, fmt::format("\"{}\"", c10::toString(debugInfo->getDType())));
   map.emplace(kInMsgSize, std::to_string(debugInfo->getInMessageSize()));
   map.emplace(kOutMsgSize, std::to_string(debugInfo->getOutMessageSize()));
-  map.emplace(
+  auto& inSplitSizes = debugInfo->getInputSplitSizes();
+  if (!inSplitSizes.empty() && inSplitSizes.size() <= kTruncatLength) {
```

Comment:
Should we still record the first kTruncatLength in the list?

Reply:
Good point, i was thinking should we keep the first 1 or 30 elements in the list or just skip recording if the length is too long. Any recommended rule to follow?

Reply:
It would make sense to record the first 30 elements, and then add a ... when the length is greater than 30. So then we can claim to show all elements up to the first 30.

Reply:
We can address this in the future, if someone is interested.